google/cloud/spanner_dbapi/cursor.py

```diff
@@ -88,6 +89,20 @@ def execute(self, sql, args=None):
 
         self._res = None
 
+        if not self._connection.autocommit:
+            if (
+                not self.transaction
+                or self.transaction.committed
+                or self.transaction.rolled_back
+            ):
+                self.transaction = self._connection.session().transaction()
+                self.transaction.begin()
+                self._connection.transactions.append(self.transaction)
+
+            self._res = self.transaction.execute_sql(sql)
+            self._itr = PeekIterator(self._res)
+            return
```

Comment:
Starting a transaction, if no yet started, or the last one is already commited/rollbacked. Then running SQL request in the current transaction.